File: Algorithms/Javascript_Sort_Algorithms/union.js
function ba(arr1,arr2){
    i =0
    j=0
    newArr = []
    //console.log("step: Before start")
    while (i< arr1.length && j < arr2.length){
        if(arr1[i] < arr2[j] && arr1[i-1] !== arr1[i] && arr1[i+1] !== arr1[i]){
            newArr.push(arr1[i])
            i++
            console.log("Pushed1: " + arr1[i])
        }
        else if(arr2[j] < arr1[i] && arr2[j-1] !== arr2[j] && arr2[j+1] !== arr2[j]){
            newArr.push(arr2[j])
            j++
            console.log("Pushed2: " + arr2[j])
        }
        else if(arr1[i] == arr2[j]){
            i++
            j++
            newArr.push(arr1[i])
            console.log("pushed3" + arr1[i])
        }
        else if(arr1[i-1] || arr1[i+1] == arr1[i]){
            i++
            console.log("step: Third if" + i)
        }
        else if(arr2[j-1] || arr2[j+1] == arr2[j]){
            j++
            console.log("step: Fourth if" + j)
        }
    } 
    while(i < arr1.length){
        if(arr1[i-1] || arr1[i+1] == arr1[i]){
            i++
        }
        else{
            newArr.push(arr1[i])
            i++
            console.log("step: push" + " "+ arr1[i])
        }
    }
    while(j < arr2.length){
        if(arr2[j-1] || arr2[j+1] == arr2[j]){
            j++
        }
        else{
            newArr.push(arr2[j])
            j++
            console.log("step: push" + " "+ arr1[i])
        }
    }
    console.log(newArr)
}

arr1 = [2,3,3,8,10]
arr2 = [1,3,4,6]
ba(arr1,arr2)

File: Algorithms/Python_Sort_Algorithms/union.py
from tkinter import W

from numpy import append


def union(arr1,arr2):
    i = 0
    j = 0
    newArr = []
    while(i < len(arr1) and j < len(arr2)):
        if arr1[i] < arr2[j] and arr1[(i-1)] is not arr1[i] and arr1[(i+1)] is not arr1[i]:
            newArr.append(arr1[i])
        elif arr2[j] < arr2[i] and arr2[(j-1)] is not arr2[j] and arr2[(j+1)] is not arr2[j]:
            newArr.append(arr2[j])
        elif arr1[i] == arr2[j] :
            i+=i
            j+=j
            newArr.append(arr1[i])
        elif arr1[i] == arr1[i-1] or arr1[i] == arr1[i+1]:
            i+=i
        elif arr2[j] == arr1[i -1] or arr2[j] == arr2[j+1]:
            j+=j

    while(i < len(arr1)):
        newArr.append(arr1[i])
        i+=i
    while(j < len(arr2)):
        newArr.append(arr2[j])
    